src/Layer.js

```diff
@@ -1,39 +1,25 @@
 import mitt from 'mitt';
-import CartoValidationError from './errors/carto-validation-error';
-import getCMIntegrator from './integrator/carto';
-import CartoMap from './integrator/Map';
-import getMGLIntegrator from './integrator/mapbox-gl';
+import util from './utils/util';
+import Viz from './Viz';
+import SourceBase from './sources/Base';
+import Renderer from './renderer/Renderer';
 import RenderLayer from './renderer/RenderLayer';
+import CartoValidationError from './errors/carto-validation-error';
 import { cubic } from './renderer/viz/expressions';
-import SourceBase from './sources/Base';
-import util from './utils/util';
 import { layerVisibility } from './constants/layer';
-import Viz from './Viz';
-
-/**
- *
- * LayerEvent objects are fired by {@link carto.Layer|Layer} objects.
- *
- * @typedef {object} LayerEvent
- * @api
- */
 
-/**
- * A loaded event is fired once the layer is firstly loaded. Loaded events won't be fired after the initial load.
- *
- * @event loaded
- * @type {LayerEvent}
- * @api
- */
+// There is one renderer per map, so the layers added to the same map
+// use the same renderer with each renderLayer
+const renderers = new WeakMap();
 
-/**
- * Updated events are fired every time that viz variables could have changed, like: map panning, map zooming, source data loading or viz changes.
- * This is useful to create external widgets that are refreshed reactively to changes in the CARTO VL map.
- *
- * @event updated
- * @type {LayerEvent}
- * @api
-*/
+function getRenderer (map, gl) {
```

Comment:
This is a private function that should be placed at the bottom (after the class) and should be `_getRenderer` instead of `getRenderer`.